### DISCUSSION TITLE
ModuleNotFoundError: No module named 'flask._compat' in Flask 3.0.0

### DISCUSSION THREAD

**Discussion by SWS-5007** [pallets/flask#5321] [Q&A] [answered]
Hi,
When I run python project by using `py manage.py` command, I am getting error below.

> Administrator@DESKTOP-BM42CU5 MINGW64 /e/tellmewhentogo_backend (main)
> $ py manage.py
> Traceback (most recent call last):
>   File "E:\tellmewhentogo_backend\manage.py", line 6, in <module>
>     from flask_script import Manager, Shell
>   File "C:\Users\HOME\AppData\Local\Programs\Python\Python310\lib\site-packages\flask_script\__init__.py", line 15, in <module>
>     from flask._compat import text_type
> ModuleNotFoundError: No module named 'flask._compat'

**Here is my environment info.**
```
Python 3.10.8
Flask 3.0.0  
Werkzeug 3.0.1
```

**And here is my ` manage.py` codebase.**
```
import os
from app import create_app, db, bcrypt, marketManager
from app.models.role import Role
from app.models.user import User
from app.models.market import Market
from flask_script import Manager, Shell
from flask_migrate import Migrate

import warnings
warnings.filterwarnings("ignore")

app = create_app(os.getenv('FLASK_CONFIG') or 'default')
manager = Manager(app)
migrate = Migrate(app, db)


def make_shell_context():
    return dict(app=app, db=db, User=User)

manager.add_command("shell", Shell(make_context=make_shell_context))

@manager.command
def runserver():
    marketManager.initialize()
    app.run()

@manager.command
def dev():
    marketManager.initialize()
    app.run(port=5001)
    
@manager.command
def test():
    import unittest
    tests = unittest.TestLoader().discover('tests')
    unittest.TextTestRunner(verbosity=2).run(tests)

@manager.command
def createDb():
    db.create_all()

   
@manager.command
def dbInit():
    market = Market('BTCUSDT')
    db.session.add(market)
    db.session.commit()

if __name__ == '__main__':
    manager.run()
```

**And here is the entire source code.**
[https://github.com/SWS-5007/TellMeWhenToGo_Python](https://github.com/SWS-5007/TellMeWhenToGo_Python)

How to fix the current error?
Please help me.

**Comment by ThiefMaster** [comment, marked as answer]
Flask-Script is dead. It has been last updated in 2017 and the repo is archived since 2020. Use the built-in flask cli (based on click) instead.

**Reply by SWS-5007** [reply]
Can you teach me the correct codebase by using built-in flask cli?
Actually, I have no experience in this part.
Best

**Reply by ThiefMaster** [reply]
I recommend reading the Flask documentation and quickstart guide. And if that project you linked to is something you did right now, stop using whatever obsolete book/tutorial/video is telling you to use flask-script.

**Reply by SWS-5007** [reply]
Okay, got it.
Thanks so much for your helps.
Best
